stark/src/merkle_tree/commitment.rs

```diff
@@ -0,0 +1,62 @@
+use super::{Error, Hash, Index, Result};
+use itertools::Itertools;
+use std::prelude::v1::*;
+
+#[derive(Clone)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub struct Commitment {
+    depth: usize,
+    hash:  Hash,
+}
+
+impl Commitment {
+    pub fn from_depth_hash(depth: usize, hash: &Hash) -> Result<Self> {
+        if depth >= (0_usize.count_zeros() as usize) {
+            // The number of leaves needs to fit `usize`
+            Err(Error::DepthOutOfRange)
+        } else {
+            Ok(Self {
+                depth,
+                hash: hash.clone(),
+            })
+        }
+    }
+
+    pub fn depth(&self) -> usize {
+        self.depth
+    }
+
+    pub fn num_leaves(&self) -> usize {
+        1_usize << self.depth
+    }
+
+    pub fn hash(&self) -> &Hash {
+        &self.hash
+    }
+
+    /// Convert leaf indices to a sorted list of unique `Index`s.
```

Comment:
Can you add "And verifies that the indices are all valid"

Reply:
I added it, but I think it goes without saying that a function validates it's input :)

Reply:
IMO, from the function name, it's not clear why sorting indices could result in an error.